File: Python/0019_remove_nth_node_from_end_of_list/0019_remove_nth_node_from_end_of_list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def removeNthFromEnd(self, head, n):
        """
        :type head: ListNode
        :type n: int
        :rtype: ListNode
        """
        if head == None or n <= 0:
            return None  
        tempHead = ListNode(0)
        tempHead.next = head
        first = tempHead
        second = tempHead
        i = 0
        while i < n:
            second = second.next
            i += 1
        while second.next != None:
            first = first.next
            second = second.next
        first.next = first.next.next  
        return tempHead.next

File: Java/0019_remove_nth_node_from_end_of_list/0019_remove_nth_node_from_end_of_list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        if (head == null || n <= 0)
          return null;   
        ListNode tempHead = new ListNode(0);
        tempHead.next = head;
        ListNode first = tempHead;
        ListNode second = tempHead;
        int i = 0;
        while (i < n) {
            second = second.next;
            i++;
        } 
        while (second.next != null) {
            first = first.next;
            second = second.next;
        }  
        first.next = first.next.next;  
        return tempHead.next;
    } 
}
